standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -180,12 +194,33 @@ impl BtcLottery {
                         return;
                     }
                 };
+                let secp = Secp256k1::new();
+                let public_key = PublicKey::from_private_key(&secp, &sk);
+                let prize_addr = Address::p2pkh(&public_key, Network::Bitcoin);
+                address_set.push(prize_addr.to_string().as_bytes().to_vec());
                 lottery_token.insert(String::from(winner_id), sk);
                 salt += 1;
             }
             self.lottery_set.insert(round_id, lottery_token);
             self.token_set.insert(round_id, round_token);
             self.round_id = round_id;
+
+            let payload = LotteryPayload::BtcAddresses { address_set };
+            let data = SendLottery {
+                payload_type: 0,
```

Comment:
Actually payload_type is not necessary as well, because payload itself already has the type encoded inside